Src/FluentAssertions/Equivalency/ConstraintEquivalencyStep.cs

```diff
@@ -0,0 +1,197 @@
+﻿using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+
+using FluentAssertions.Common;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Equivalency
+{
+    public class ConstraintEquivalencyStep : IEquivalencyStep
+    {
+        public bool CanHandle(IEquivalencyValidationContext context, IEquivalencyAssertionOptions config)
+        {
+            return typeof(Constraint).IsAssignableFrom(config.GetExpectationType(context));
+        }
+
+        public bool Handle(IEquivalencyValidationContext context, IEquivalencyValidator parent, IEquivalencyAssertionOptions config)
+        {
+            var subject = (Constraint)context.Subject;
+            var expectation = (Constraint)context.Expectation;
+
+            var selectedMembers = GetMembersFromExpectation(context, config)
+                .ToDictionary(member => member.Name);
+
+            if (selectedMembers.ContainsKey("ConstraintName"))
```

Comment:
❓ Will this work in all language-specific versions of .NET?

Reply:
I'm not sure why it wouldn't -- isn't that member discovered via reflection, and _required_ to match no matter what programming language or human language is at the top of the stack? Maybe I'm missing something?

Reply:
Let's assume it is.